### PR TITLE
qa: Remove need to handle the network thread in tests

### PR DESCRIPTION
**Pull request by maflcko**
This simplifies test writing by removing the need to handle the network thread in tests. E.g. start thread, join thread, restart thread mid-test, adding p2p connections at the "right" time, ...

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Please review the refactoring in #13512 first.

**Comment by jnewbery** [comment]
Big concept ACK!

`feature_block.py` currently failing on Travis.

File: test/functional/README.md [hunk 1]
```diff
@@ -71,23 +71,23 @@ P2P messages. These can be found in the following source files:
 
 #### Using the P2P interface
 
+- Set `setup_mininode_event_loop = True` in `set_test_params()` to spawn a thread
```
Comment (jnewbery) [inline test/functional/README.md]: Would it be possible to start the network thread for all tests, so it's not necessary for the test writer to even perform this step?

File: test/functional/README.md [hunk 2]
```diff
@@ -71,23 +71,23 @@ P2P messages. These can be found in the following source files:
...
+that runs a callback-based event loop to communicate on the P2P interface with a
+node.
+
 - `mininode.py` contains all the definitions for objects that pass
 over the network (`CBlock`, `CTransaction`, etc, along with the network-level
 wrappers for them, `msg_block`, `msg_tx`, etc).
 
 - P2P tests have two threads. One thread handles all network communication
-with the bitcoind(s) being tested (using python's asyncore package); the other
+with the bitcoind(s) being tested; the other
```
Comment (jnewbery) [inline test/functional/README.md]: nit: Join lines

File: test/functional/feature_nulldummy.py [hunk 3]
```diff
@@ -15,7 +15,7 @@
 
 from test_framework.test_framework import BitcoinTestFramework
 from test_framework.util import *
-from test_framework.mininode import CTransaction, network_thread_start
+from test_framework.mininode import CTransaction
```
Comment (jnewbery) [inline test/functional/feature_nulldummy.py]: Can be changed to `from test_framework.messages import CTransaction` :)

File: test/functional/test_framework/mininode.py [hunk 4]
```diff
@@ -13,11 +13,10 @@
 P2PInterface: A high-level interface object for communicating to a node over P2P
 P2PDataStore: A p2p interface class that keeps a store of transactions and blocks
               and can respond correctly to getdata and getheaders messages"""
-import asyncore
+import asyncio
```
Comment (jnewbery) [inline test/functional/test_framework/mininode.py]: :tada:

File: test/functional/rpc_blockchain.py [hunk 5]
```diff
@@ -41,14 +41,14 @@
 )
 from test_framework.mininode import (
     P2PInterface,
```
Comment (jnewbery) [inline test/functional/rpc_blockchain.py]: nit: Can join to a single line

**Review by jnewbery — COMMENTED** [review]
Cursory review. A few nits inline.

Will do full review once #13512 is merged.

File: test/functional/README.md (hunk 1 above)
Comment (MarcoFalke) [reply test/functional/README.md]: Good point. I guess there is no overhead in running an event loop with no events.

File: test/functional/feature_nulldummy.py (hunk 3 above)
Comment (MarcoFalke) [reply test/functional/feature_nulldummy.py]: ;)

File: test/functional/README.md (hunk 2 above)
Comment (MarcoFalke) [reply test/functional/README.md]: md renderer will take care of this ;)

**Comment by MarcoFalke** [comment]
aaaaddressed @jnewbery feedback in aaaa4a4b17506f526bebee65b0ea930289f79016

**Review by jnewbery — COMMENTED** [review]
Tested ACK aaaa4a4b17506f526bebee65b0ea930289f79016. 

Looks great Marco. Thanks for doing this. A few nits inline - feel free to ignore.

**Comment by MarcoFalke** [comment]
Force pushed to address feedback

**Comment by laanwj** [comment]
utACK fa87da2f172ae2e6dc15e9ed156a3564a8ecfbdd

**Pull request merged**